src/backend/base/langflow/components/vectorstores/cratedb.py

```diff
@@ -0,0 +1,90 @@
+import typing as t
+
+from langchain_cratedb import CrateDBVectorStore
+
+from langflow.base.vectorstores.model import LCVectorStoreComponent, check_cached_vector_store
+from langflow.helpers import docs_to_data
+from langflow.io import HandleInput, IntInput, SecretStrInput, StrInput
+from langflow.schema import Data
+
+
+class CrateDBVectorStoreComponent(LCVectorStoreComponent):
+    display_name = "CrateDBVector"
+    description = "CrateDB Vector Store with search capabilities"
+    name = "cratedb"
+    icon = "cpu"
```

Comment:
We haven't been able to derive from `PGVectorStoreComponent`, because of an admonition about an import woe, i.e. it failed because of a circular import error.

Do you have any suggestions how to make it work? For example, there might be a different adapter which derives from another one already?

Reply:
On the other hand, it is also not much code that is being duplicated, and this is not a blocking issue, but rather a code smell.

Reply:
Sounds good. You could add an icon for CrateDB and use that instead of a generic one. 

Here's a PR that does it for ScrapeGraph: https://github.com/langflow-ai/langflow/pull/5903